packages/workflow-core/src/lib/workflow-runner.test.ts

```diff
@@ -1,34 +1,340 @@
-import type { MachineConfig } from 'xstate';
-import { expect, test } from 'vitest';
+import { beforeEach, describe, expect, it, test } from 'vitest';
 import { WorkflowRunner } from './workflow-runner';
 
-test('restore workflow test', () => {
-  const userId = '123456';
+const DEFAULT_PAYLOAD = { payload: { some: 'payload' } };
 
-  const simpleMachine: MachineConfig<any, any, any> = {
-    id: 'toggle',
-    initial: 'inactive',
-    states: {
-      inactive: { on: { TOGGLE: 'active' } },
-      active: { on: { TOGGLE: 'inactive' } },
+const SINGLE_STATE_MACHINE_DEFINITION = {
+  initial: 'initial',
+  states: {
+    initial: {
+      on: { EVENT: 'initial' },
     },
-  };
+  },
+};
 
-  const service = new WorkflowRunner({
-    definition: simpleMachine,
-    workflowContext: {
-      machineContext: {
-        entityId: userId,
-      },
+const TWO_STATES_MACHINE_DEFINITION = {
+  initial: 'initial',
+  states: {
+    initial: {
+      on: { EVENT: 'final' },
     },
-    extensions: {
-      statePlugins: [],
+    final: {
+      on: { EVENT: 'initial' },
     },
+  },
+};
+
+function EventCollectingWorkflow(args) {
```

Comment:
Thoughts about renaming it to something like `createEventCollectingWorkflow`? Would make it clearer that the function returns something and more or less what it does. The current capitalization might make it easy to confuse with a class.